src/r1cs/verifier.rs

```diff
@@ -317,7 +311,11 @@ impl<'a, 'b> Verifier<'a, 'b> {
     }
 
     /// Consume this `VerifierCS` and attempt to verify the supplied `proof`.
-    pub fn verify(mut self, proof: &R1CSProof) -> Result<(), R1CSError> {
+    pub fn verify(
+        mut self,
+        proof: &R1CSProof,
+        bp_gens: &'b BulletproofGens,
```

Comment:
Do we need a named lifetime parameter here? The `&BulletproofGens` reference isn't retained anywhere, so I think it just needs to outlive the `verify` function (unlike in the old code, where we needed it to outlive the entire verifier).